### PR TITLE
Fran preferred payment type and preferred giving type does not get updated

### PR DESCRIPTION
**Pull request by jfescobar18**
https://trello.com/c/xjgD8F4e

### CODE REVIEW

File: CmsWeb/Areas/OnlineReg/Models/ManageGivingModel.cs [hunk 1]
```diff
@@ -21,32 +22,14 @@
 namespace CmsWeb.Areas.OnlineReg.Models
 {
     [Serializable]
-    public class ManageGivingModel
+    public class ManageGivingModel : IDbBinder
     {
+        public CMSDataContext CurrentDatabase { get; set; }
```
Comment (brondavies) [inline CmsWeb/Areas/OnlineReg/Models/ManageGivingModel.cs]: Add attribute `[NonSerialized]`

File: CmsWeb/Areas/OnlineReg/Models/OnlineReg/PaymentForm.cs [hunk 2]
```diff
@@ -13,11 +14,12 @@
 
 namespace CmsWeb.Areas.OnlineReg.Models
 {
-    public class PaymentForm
+    public class PaymentForm : IDbBinder
     {
+        public CMSDataContext CurrentDatabase { get; set; }
```
Comment (brondavies) [inline CmsWeb/Areas/OnlineReg/Models/OnlineReg/PaymentForm.cs]: Add attribute `[NonSerialized]`

File: CmsWeb/Areas/OnlineReg/Models/OnlineReg/PaymentForm.cs [hunk 3]
```diff
@@ -147,6 +148,11 @@ public string FullName()
             return n;
         }
 
+        public PaymentForm()
```
Comment (brondavies) [inline CmsWeb/Areas/OnlineReg/Models/OnlineReg/PaymentForm.cs]: The parameterless constructor should only be used by model binding.  This should be an empty method with the `[Obsolete(Errors.ModelBindingConstructorError, error: true)]` attribute.  Then, add a constructor that takes a `CMSDataContext` and set that to `CurrentDatabase`.  Anywhere the compiler stops you from using the parameterless constructor, change it to pass in the `CMSDataContext`

File: CmsWeb/Areas/Org/Models/OrgSearchModel.cs [hunk 4]
```diff
@@ -45,7 +45,7 @@ public OrgSearchModel(CMSDataContext db)
 
         public OrgSearchModel()
         {
-            CurrentDatabase = CurrentDatabase ?? DbUtil.Db;
+            CurrentDatabase = CurrentDatabase ?? CMSDataContext.Create(Util.Host);
```
Comment (brondavies) [inline CmsWeb/Areas/Org/Models/OrgSearchModel.cs]: Same thing here - use the model binding pattern described above.  Let me know if you have any questions

File: CmsWeb/Areas/OnlineReg/Models/OnlineReg/PaymentForm.cs [hunk 5]
```diff
@@ -48,11 +50,14 @@ public class PaymentForm : IDbBinder
...
-        public string URL { get; set; }        
-        public CMSDataContext CurrentDatabase { get; set; }
+        public string URL { get; set; }
 
-        [Obsolete(Errors.ModelBindingConstructorError, true)]
-        public PaymentForm() { }
+        [Obsolete(Errors.ModelBindingConstructorError, error: true)]
+        public PaymentForm()
+        {
+
```
Comment (josearmandohf) [inline CmsWeb/Areas/OnlineReg/Models/OnlineReg/PaymentForm.cs]: please fix the code factor extra blank spaces

File: CmsWeb/Areas/OnlineReg/Models/OnlineReg/Constructor.cs [hunk 6]
```diff
@@ -36,14 +41,16 @@ private void Init()
...
+        public OnlineRegModel(CMSDataContext db)
+        {            
+            CurrentDatabase = db;
+        }
 
         [Obsolete(Errors.ModelBindingConstructorError, true)]
-        public OnlineRegModel() { Init(); }
-
-        public OnlineRegModel(CMSDataContext db)
+        public OnlineRegModel()
```
Comment (josearmandohf) [inline CmsWeb/Areas/OnlineReg/Models/OnlineReg/Constructor.cs]: some blank spaces here

File: CmsWeb/Areas/OnlineReg/Models/OnlineReg/Constructor.cs [hunk 7]
```diff
@@ -36,14 +41,16 @@ private void Init()
...
         {
-            CurrentDatabase = db;
+
```
Comment (josearmandohf) [inline CmsWeb/Areas/OnlineReg/Models/OnlineReg/Constructor.cs]: some blank spaces here

File: CmsWeb/Areas/Dialog/Models/OrganizationModel.cs [hunk 8]
```diff
@@ -34,6 +38,12 @@ public NewOrganizationModel(int? id, bool displayCopySettings = false)
         public NewOrganizationModel()
```
[the remark this answers is not in the document]
Comment (jfescobar18) [reply CmsWeb/Areas/Dialog/Models/OrganizationModel.cs]: Fixed

File: CmsWeb/Areas/OnlineReg/Models/OnlineReg/Constructor.cs (hunk 6 above)
Comment (jfescobar18) [reply CmsWeb/Areas/OnlineReg/Models/OnlineReg/Constructor.cs]: All these Codefactor marks are from old version, all are resolved.

**Pull request merged**